lib/manager/terraform/extract.ts

```diff
@@ -31,33 +35,55 @@ export function getTerraformDependencyType(
 
 export function extractPackageFile(content: string): PackageFile | null {
   logger.trace({ content }, 'terraform.extractPackageFile()');
-  if (!content.includes('module "') && !content.includes('provider "')) {
+  if (
+    !content.includes('module "') &&
+    !content.includes('provider "') &&
+    !content.includes('required_providers ')
+  ) {
     return null;
   }
   const deps: PackageDependency[] = [];
   try {
     const lines = content.split('\n');
     for (let lineNumber = 0; lineNumber < lines.length; lineNumber += 1) {
       let line = lines[lineNumber];
-      const terraformDependency = /^(module|provider)\s+"([^"]+)"\s+{\s*$/.exec(
+      const terraformDependency = /^\s*(module|provider|required_providers)\s+("([^"]+)"\s+)?{\s*$/.exec(
         line
       );
       if (terraformDependency) {
         logger.trace(`Matched ${terraformDependency[1]} on line ${lineNumber}`);
-        const tfDepType: TerraformDependencyTypes = getTerraformDependencyType(
-          terraformDependency[1]
-        );
-        const dep: PackageDependency = {
-          managerData: {
-            moduleName: terraformDependency[2],
-            terraformDependencyType: tfDepType,
-          },
-        };
+        const tfDepType = getTerraformDependencyType(terraformDependency[1]);
+
         if (tfDepType === TerraformDependencyTypes.unknown) {
           /* istanbul ignore next */ logger.trace(
             `Could not identify TerraformDependencyType ${terraformDependency[1]} on line ${lineNumber}.`
           );
+        } else if (tfDepType === TerraformDependencyTypes.required_providers) {
+          do {
+            const dep: PackageDependency = {
+              managerData: {
+                terraformDependencyType: tfDepType,
+              },
+            };
+
+            lineNumber += 1;
+            line = lines[lineNumber];
+            const kvMatch = /^\s*([^\s]+)\s+=\s+"([^"]+)"\s*$/.exec(line);
```

Comment:
Please move regex to module scope constant. Also named capture groups are easier to read.